src/Models/Market.php

```diff
@@ -52,6 +59,21 @@ protected static function boot()
             }
         });
 
+        static::updating(function ($market) {
+            $other_market = Market::where('slug', $market->slug)->where('id', "<>", $market->id)->first();
```

Comment:
Same here -- silently not persisting the change can lead to errors.  Better as an exception.